deepcell_spots/applications/polaris.py

```diff
@@ -23,227 +23,142 @@
-        label_image = self._resize_output(output_images, image.shape)
+        valid_compartments = ['cytoplasm', 'nucleus', 'None']
+        if segmentation_compartment not in valid_compartments:
+            raise ValueError('Invalid compartment supplied: {}. '
+                             'Must be one of {}'.format(segmentation_compartment,
+                                                        valid_compartments))
 
-        # Postprocess predictions to create label image
-        predicted_spots = self._postprocess(label_image, **postprocess_kwargs)
-
-        return predicted_spots
+        if segmentation_compartment == 'cytoplasm':
+            self.segmentation_app = CytoplasmSegmentation(model=segmentation_model)
+            self.segmentation_app.preprocessing_fn = histogram_normalization
+            self.segmentation_app.postprocessing_fn = deep_watershed
+        elif segmentation_compartment == 'nucleus':
+            self.segmentation_app = NuclearSegmentation(model=segmentation_model)
+        else:
+            self.segmentation_app = None
 
     def predict(self,
-                image,
-                batch_size=4,
+                spots_image,
+                segmentation_image=None,
                 image_mpp=None,
-                pad_mode='reflect',
-                preprocess_kwargs=None,
-                postprocess_kwargs=None,
-                threshold=0.9):
-        """Generates a list of coordinate spot locations of the input
-        running prediction with appropriate pre and post processing
-        functions.
+                spots_threshold=0.95,
+                spots_clip=False):
+        """Generates prediction output consisting of a labeled cell segmentation image,
+        detected spot locations, and a dictionary of spot locations assigned to labeled
+        cells of the input.
+
         Input images are required to have 4 dimensions
-        ``[batch, x, y, channel]``.
+        ``[batch, x, y, channel]``. Channel dimension should be 2.
+
         Additional empty dimensions can be added using ``np.expand_dims``.
+
         Args:
             image (numpy.array): Input image with shape
                 ``[batch, x, y, channel]``.
-            batch_size (int): Number of images to predict on per batch.
             image_mpp (float): Microns per pixel for ``image``.
-            pad_mode (str): The padding mode, one of "constant" or "reflect".
-            preprocess_kwargs (dict): Keyword arguments to pass to the
-                pre-processing function.
-            postprocess_kwargs (dict): Keyword arguments to pass to the
-                post-processing function.
+            cytoplasm_channel (int): Value should be 0 or 1 depending on the channel
+                containing the images for cell segmentation. Defaults to 0.
+            spots_channel (int): Value should be 0 or 1 depending on the channel
+                containing the images for spot detection. Defaults to 1.
```

Comment:
Update docstring to match new pattern of passing in an image per channel instead of specifying the channel index